api/redhatopenshift/HcpCluster/hcpCluster-models.tsp

```diff
@@ -260,12 +257,43 @@ model PlatformProfile {
 
   /** Specifies whether subnets are pre-attached with an NSG */
   preconfiguredNsgs: boolean;
+}
 
-  /** The id of the disk encryption set to be used for etcd.
-   * Configure this when `etcdEncryption` is set to true
-   * Is used the https://learn.microsoft.com/en-us/azure/storage/common/customer-managed-keys-overview
-   */
-  etcdEncryptionSetId?: string;
+/** etcd specific configuration */
+model EtcdProfile {
+  encryption?: EtcdEncryptionProfile;
+}
+
+/** etcd encryption specific configuration */
+model EtcdEncryptionProfile {
+  dataEncryption?: EtcdDataEncryption;
+}
+
+/** etcd data-level encryption specific configuration.
+ * At least one of its attributes is required. At most
+ * one attribute can be specified.
+ */
+model EtcdDataEncryption {
+  kms?: EtcdDataEncryptionKms;
+}
+
+/** etcd KMS-based data-level encryption specific configuration */
+model EtcdDataEncryptionKms {
+	/** The name of the Azure Key Vault vault containing encryption
+	 * keys used to perform etcd data-level encryption.
+	 */
+  keyVaultName: string;
```

Comment:
To be decided is this has to belong to specific subscription and/or  resource group name or we want flexibility around any arbitrary resource id. The decision on this might require changes on hypershift side itself